sdk/identity/identity/test/internal/node/managedIdentityCredential.spec.ts

```diff
@@ -299,6 +299,34 @@ describe("ManagedIdentityCredential", function () {
     );
   });
 
+  it("IMDS MSI accepts a custom set of retries, even when client Id is passed through the first parameter", async function () {
+    const { error } = await testContext.sendCredentialRequests({
+      scopes: ["scopes"],
+      credential: new ManagedIdentityCredential("errclient", {
+        retryOptions: {
+          maxRetries: 4,
+        },
+      }),
+      insecureResponses: [
+        // Any response on the ping request is fine, since it means that the endpoint is indeed there.
+        createResponse(503, {}, { "Retry-After": "2" }),
```

Comment:
This is a pretty clever way of testing that the setting took effect.

Reply:
Thank you 🙏❤️